src/proto/FunctionRpc.proto

```diff
@@ -315,9 +327,18 @@ message TypedData {
     CollectionString collection_string = 9;
     CollectionDouble collection_double = 10;
     CollectionSInt64 collection_sint64 = 11;
+    SharedMemoryData shared_memory_data = 12;
```

Comment:
So something like:

```
// Used to describe a given binding on invocation
message ParameterBinding {
  // Name for the binding
  string name = 1;

  oneof param_data {
    // Data for the binding
    TypedData data = 2;

    // Data for shared memory
    RpcSharedMemoryMapInfo memory_map_info = 3;
  }
}
```

This is because we either send the data through RPC itself in which case `ParameterBinding.param_data.data` will be populated or we send data through shared memory in which case `ParameterBinding.param_data.memory_map_info` will be populated. This way the worker can determine how to obtain the data sent by the host.

Is this more or less what you are pointing at?

Reply:
Yes this is exactly what i had in mind. But not sure if we can introduce `oneof` in a backward compatible way.